filebeat/harvester/source/source.go

```diff
@@ -0,0 +1,17 @@
+package source
+
+import (
+	"io"
+	"os"
+)
+
+type LogSource interface {
+	io.ReadCloser
+	Name() string
+}
+
+type FileSource interface {
+	LogSource
+	Stat() (os.FileInfo, error)
+	Continuable() bool // can we continue processing after EOF?
```

Comment:
I don't like this Continuable. It's here due to stdin and log-file tailing using same reader implementation. A custom reader pipeline for files and stdin each will not require this 'Continuable' callback. Custom readers will make `FileSource` interface obsolete.

Reason for FileSource to exist (instead of plain LogSource) is requirement of `Stat()` in harvester when dealing with truncated files. I think with recent refactorings, Stat() is not required by harvester loop anymore.